tests/swap.rs

```diff
@@ -1998,19 +1999,43 @@ async fn run_punish_swap_kill_bob_before_monero_funding(
         .generate_to_address(3, &reusable_btc_address())
         .unwrap();
 
-    info!("generated 20 bitcoin blocks");
+    info!("generated 3 bitcoin blocks");
 
-    // run until the FailurePunish is received
+    // run until the FailurePunish is received for Alice
     retry_until_finish_transition(
         cli_alice_progress_args.clone(),
         "Failure Punish".to_string(),
     )
     .await;
 
+    info!("restoring Bob");
+    // restore Bob from checkpoint
+    launch_farcasterd_maker();
+
+    // wait a bit for all of Bob's daemons to start
+    tokio::time::sleep(time::Duration::from_secs(2)).await;
+
+    restore_checkpoint(swap_id, data_dir_bob.clone());
+
+    // wait a bit for Bob to get up to speed
+    tokio::time::sleep(time::Duration::from_secs(5)).await;
+
+    bitcoin_rpc
+        .generate_to_address(10, &reusable_btc_address())
+        .unwrap();
```

Comment:
It's a bit weird that this is required. Shouldn't Bob automatically go through the transitions, since they are all already confirmed and final?

Reply:
34b53b9b8ad9420b761e3840d40d15417d419770